packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -1713,30 +1734,100 @@ export class SnapController extends BaseController<
     }
 
     try {
-      const { sourceCode } = await this.#add({
-        origin,
-        id: snapId,
-        location,
-      });
+      try {
+        await this.#add({
+          origin,
+          id: snapId,
+          location,
+        });
+      } catch (error) {
+        this.#updateApproval(permissionsApproval.id, {
```

Comment:
To avoid nested try catch I think we catch this in the existing catch block inside `#add` instead.